atmcfg/atmcfg.go

```diff
@@ -72,11 +78,45 @@ func setDisabledByShardName(out *opsmngr.AutomationConfig, name string, disabled
 	}
 }
 
+func setDisableProcessByProcessNameAndPort(out *opsmngr.AutomationConfig, processName string, processPortNum int, disabled bool) {
+	newDeploymentAuthMechanisms(out)
+	for k, p := range out.Processes {
+		if p.Hostname == processName && p.Args26.NET.Port == processPortNum {
+			out.Processes[k].Disabled = disabled
+		}
+	}
+}
+
 // Shutdown disables all processes of the given cluster name.
 func Shutdown(out *opsmngr.AutomationConfig, name string) {
 	setDisabledByClusterName(out, name, true)
 }
 
+// ShutdownProcess disables processes. Processes are provided in the format {"hostname:port","hostname2:port2"}.
+func ShutdownProcess(out *opsmngr.AutomationConfig, processes []string) error {
+	for _, hostnameAndPort := range processes {
+		hostname, port, err := splitHostnameAndPort(hostnameAndPort)
+		if err != nil {
+			return err
+		}
+		setDisableProcessByProcessNameAndPort(out, hostname, port, true)
+	}
```

Comment:
this is a bit wasteful, you'll have to iterate over the whole list of processes in the config for each process in the args

consider creating a map based on the args, then do a single for loop that checks the `fmt.Stringf("%s:%d", p.Hostname , p.Args26.NET.Port` is present on the map this also removes the need to split, and brings a question up to you, what happens if elements in the map are not in the process list?

(also keep in mind my previous comment, you want to look at the cluster name as well)

Reply:
I don't understand why we need to look at the cluster name here, at the end we have all the information that we need to find the process that we want without the need of the cluster name.

Reply:
In addition we make the cluster name required for the command even if we do not need it which I personally don’t like

Reply:
the reason is required is to prevent shutting down the wrong process, you ask for two data points instead of one

this is as saying for cluster A shutdown process B, if process B is not part of cluster A maybe the user should double check what they are doing and links to my point, what happens if a user gives a process name that doesn't exists? we should make them double check

Reply:
ah okay. My idea was to return an error if the user provides a process that does not exist

Reply:
agree on return an error and why is important to make sure the process not only exists but belongs to the cluster you asked for